lib/assertions/matches-selector.ts

```diff
@@ -0,0 +1,15 @@
+export default function matchesSelector(
+  targetSelector: string | Element,
+  compareSelector: string
+): Array<number> {
+  let targetElements: Array<Element> =
+    typeof targetSelector === 'string'
+      ? Array.from(document.querySelectorAll(targetSelector))
+      : [targetSelector];
+
+  let failures = targetElements.reduce(function(acc, element) {
+    return element.matches(compareSelector) ? acc : acc + 1;
+  }, 0);
```

Comment:
```js
let failures = targetElement.filter(it => it.matches(compareSelector)).length > 0;
```
👆 seems a little easier to read to me

Reply:
agreed, but to count failures, we filter the negative results, making it `!it.matches(...)`.